app/questionnaire/router.py

```diff
@@ -299,34 +309,44 @@ def _get_return_to_for_grand_calculated_summary(
         grand_calculated_summary_section: str = (
             self._schema.get_section_id_for_block_id(return_to_block_id)  # type: ignore
         )
+        list_item_id = location.list_item_id or return_to_list_item_id
+        list_name = (
+            self._list_store.get_list_name_for_list_item_id(list_item_id)
+            if list_item_id
+            else None
+        )
         if grand_calculated_summary_section != section_key.section_id:
             # the grand calculated summary is in a different section which will have a different routing path
             # but does not go to it unless the section is enabled and the current section is complete
             if (
-                not self._progress_store.is_section_complete(
-                    SectionKey(section_key.section_id)
-                )
+                not self._progress_store.is_section_complete(section_key)
                 or grand_calculated_summary_section not in self.enabled_section_ids
             ):
                 return None
```

Comment:
Thanks for spotting this - have added an extra question depending on that first calculated summary so this can be manually & functionally tested and passed it to that extra method

Reply:
We need to look at this wider as anchor/return to answer id is another culprit. Once we add a `ReturnLocation` it should because easily to spot and not miss.